invenio_app_ils/circulation/search.py

```diff
@@ -88,6 +89,9 @@ def _default_parser(qstr=None):
     query_string = request.values.get('q', '')
     query = _default_parser(qstr=query_string)
 
+    if not current_user.is_authenticated:
+        raise UnauthorizedSearchError(query_string, patron_pid=None)
```

Comment:
We can possible skip the `patron_pid=None`

Reply:
Fixed now! I also updated `UnauthorizedSearchError` and the error tests.